SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkType.swift

```diff
@@ -31,13 +35,84 @@ func initializeNetworkTypeMonitoring() {
             if path.status == .satisfied {
                 if path.usesInterfaceType(.wifi) {
                     hostConnectionType = "wifi"
+                    hostConnectionSubtype = getNetworkInfo()
                 } else if path.usesInterfaceType(.cellular) {
                     hostConnectionType = "cell"
+                    hostConnectionSubtype = getNetworkInfo()
+                    networkDetector()
                 } else {
                     hostConnectionType = nil
+                    hostConnectionSubtype = nil
+                }
+            }
+        }
+    }
+}
+
+func networkDetector() {
+    if #available(iOS 12.0, *) {
+        if let carrier = networkInfo.serviceSubscriberCellularProviders {
+               carrier.forEach { (_, value) in
+                if value.mobileCountryCode != nil {
+                    hostConnectionName = value.carrierName
                 }
             }
         }
+    } else {
+        let networkCarrier = networkInfo.subscriberCellularProvider
+        let networkCarrierName = networkCarrier?.carrierName
+        if networkCarrierName != nil {
+            hostConnectionName = networkCarrierName
+        }
     }
+}
+
+ func getNetworkInfo() -> String? {
 
+     if #available(iOS 12.0, *) {
+         guard let dict = networkInfo.serviceCurrentRadioAccessTechnology else {
+             return nil
+         }
+         if dict.count != 0 {
```

Comment:
Is it possible here are multiple networks in use at the same time here? 🤔

Reply:
Dictionary containing the current radio access technology registered to each service.
1.Physical SIM card.
2.eSIM.